lib/phoenix_live_view/js.ex

```diff
@@ -218,10 +218,120 @@ defmodule Phoenix.LiveView.JS do
 
   defimpl Phoenix.HTML.Safe, for: Phoenix.LiveView.JS do
     def to_iodata(%Phoenix.LiveView.JS{} = js) do
-      Phoenix.HTML.Engine.html_escape(Phoenix.json_library().encode!(js.ops))
+      js
+      |> JS.to_encodable()
+      |> Phoenix.json_library().encode!()
+      |> Phoenix.HTML.Engine.html_escape()
     end
   end
 
+  if Code.ensure_loaded?(Jason.Encoder) do
+    defimpl Jason.Encoder, for: Phoenix.LiveView.JS do
+      def encode(%Phoenix.LiveView.JS{} = js, opts) do
+        Jason.Encode.list(JS.to_encodable(js), opts)
+      end
+    end
+  end
+
+  if Code.ensure_loaded?(JSON.Encoder) do
+    defimpl JSON.Encoder, for: Phoenix.LiveView.JS do
+      def encode(%Phoenix.LiveView.JS{} = js, encoder) do
+        JSON.Encoder.encode(JS.to_encodable(js), encoder)
+      end
+    end
+  end
```

Comment:
I think we either define the protocol or we export `to_encodable`, in my mind there is no reason to define both. So we need to pick one.

@chrismccord @SteffenDE preferences?